Sources/TuistGenerator/Generator/ProjectGroups.swift

```diff
@@ -25,13 +25,14 @@ enum ProjectGroupsError: FatalError, Equatable {
 class ProjectGroups {
     // MARK: - Attributes
 
-    let main: PBXGroup
+    private let main: PBXGroup // main should be accessed from buildMain
```

Comment:
What about using [property wrappers](https://nshipster.com/propertywrapper/) that we can use with these properties:

```swift
@SortedPBXGroup let main: PBXGroup
@SortedPBXGroup let frameworks: PBXGroup
@SortedPBXGroup let playgrounds: PBXGroup
```

That way you can move the logic from the `PBXGroupSorter` into the annotation.

Reply:
I've added `@SortedPBXGroup let main: PBXGroup` but left the others as they are as I believe they are all sorted when main is sorted. I realise that this is inconsistent but to me it seems that we do not need to sort the others unnecessarily when they are accessed.